multimodal/src/autogluon/multimodal/models/mmdet_image.py

```diff
@@ -341,8 +344,50 @@ def get_layer_ids(
         Basically, id gradually increases when going from the output end to
         the input end. The layers defined in this class, e.g., head, have id 0.
 
-        Setting all layers as the same id 0 for now.
-        TODO: Need to investigate mmdetection's model definitions
+        Returns
+        -------
+        A dictionary mapping the layer names (keys) to their ids (values).
+        """
+
+        # two stage lr: backbone v.s. else, or head v.s. else
+        if self.backbone_layers:
+            return self.get_layer_ids_by_backbone()
+        else:
+            return self.get_layer_ids_by_head()
+
+    def get_layer_ids_by_backbone(
+        self,
+    ):
+        """
+        Assign an id to each layer. Layer ids will be used in layer-wise lr decay.
+        Basically, id gradually increases when going from the output end to
+        the input end. The layers defined in this class, e.g., head, have id 0.
+
+        Currently only head to 0 others to 1.
+
+        Returns
+        -------
+        A dictionary mapping the layer names (keys) to their ids (values).
+        """
+        name_to_id = {}
+        backbones = self.backbone_layers
+
+        for n, _ in self.named_parameters():
+            name_to_id[n] = 0
+            for backbone in backbones:
+                if backbone in n:
+                    name_to_id[n] = 1
```

Comment:
Wrong logic of using a nested for loop?

Reply:
If any backbone pattern is included in parameter names, it's considered as a backbone layer.